### DISCUSSION TITLE
read-line function

### DISCUSSION THREAD

**Discussion by Chemaclass** [phel-lang/phel-lang#154] [Q&A] [answered]
I didn't find any function to read from the console, for example. 

I think it would be nice to have a read-line function, something like `(read-line)`
- Reference: Clojure https://clojuredocs.org/clojure.core/read-line

Extra: 
We could even go one step forward and think if it makes sense to work on an io module as well
https://github.com/clojure/clojure/blob/master/src/clj/clojure/java/io.clj

What do you think, @jenshaase? 🤔

**Comment by jenshaase** [comment, marked as answer]
Regarding `(read-line)`:

You can use `(php/readline)` if you have `ext-readline` installed. I will not wrap this function in the core library because it requires a extension.

Regarding an io:

In the clojure / Java world it makes sense to wrap the `java.io` library because it is well defined and used very often in a lot of code. In PHP the io utils are much simpler. I would recommend to use them directly (e.g. `(php/file_get_contents "path/to/file")` or `(php/file_put_contents "path/to/file" "content")`). However, if you have a cool idea on how to make a very good io library you can just create a library for it in an extra repository.

In general I want to keep the core library of Phel small. I would like to focus on stuff that requires as common specification (like the request and response object) or very general functions and macros. Moving libraries in extra repositories has also the benefit that new versions can be published independently from the compiler.

**Reply by Chemaclass** [reply]
Right! I think what I was missing was some examples and some help about how to use some basic IO, but you just told me with your examples. `(php/readline)` is more than sufficient to accomplish this. I like a lot that you can use any PHP functions by `(php/*)` ⚡
